src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Android.cs

```diff
@@ -27,16 +27,18 @@ public static void VerifyPackageInfo()
         }
 
         public static SecurityStatusPal AcceptSecurityContext(
+            SslStream.JavaProxy sslStreamProxy,
             ref SafeFreeCredentials credential,
             ref SafeDeleteSslContext? context,
             ReadOnlySpan<byte> inputBuffer,
             ref byte[]? outputBuffer,
             SslAuthenticationOptions sslAuthenticationOptions)
         {
-            return HandshakeInternal(credential, ref context, inputBuffer, ref outputBuffer, sslAuthenticationOptions);
+            return HandshakeInternal(sslStreamProxy, credential, ref context, inputBuffer, ref outputBuffer, sslAuthenticationOptions);
```

Comment:
similar as above. If you make the proxy part of `SafeFreeCredentials` then you don't need to change the prototypes. This would be similar to Windows. The credentials are useless now on Linux & macOS but we drag them along anyway to keep the structure similar.

Reply:
I experimented with the code structure a bit and I tried to put that proxy somewhere so we don't need to modify the signature. The `SafeFreeCredentials` options seems quite hacky so I tried to pass it via a "stub" context. I'm not happy with that solution either. I'll give it one more try next week.